Python/thread_pthread.h

```diff
@@ -152,6 +153,26 @@ PyThread__init_thread(void)
  * Thread support.
  */
 
+typedef struct {
+    void (*func) (void *);
+    void *arg;
+    bool copied;
```

Comment:
Thank you for your review.

Why does the flag need to be volatile?  The release fence should be sufficient to ensure that the compiler doesn't do anything funny even if it does IPA.  As for busy waiting vs having a mutex lock, I chose the former as a tradeoff since the synchronization requirement is fairly minor and didn't seem to justify the additional context switch.  I don't have a strong opinion either way so if a mutex is desirable I'll change my patch accordingly.

malloc won't really solve anything since it will also require synchronization to free the block, unless you're proposing leaking that block, i.e. not freeing it at all.

I just noticed that the CI failed on the patch, I'll fix it up and re-post.

Reply:
> Why does the flag need to be volatile?

You check the flag in the parent thread. Unless the compiler understands that the structure escapes into another thread, it may optimize the check way. I've not verified it, but the reason of the hang in the CI might be just that.

> malloc won't really solve anything since it will also require synchronization to free the block

I meant malloc + ownership transfer to the child, so that the child must free the block when it's done. It's the same as in `PyThread_start_new_thread` in `Python/thread_nt.h`.

Reply:
Yeah you're right, the malloc way can actually be easier, I just didn't see it at first.  Updated patch coming up.